src/main/java/org/opensearch/security/privileges/ResourceAccessEvaluator.java

```diff
@@ -98,6 +99,7 @@ public boolean shouldEvaluate(ActionRequest request) {
             ConfigConstants.OPENSEARCH_RESOURCE_SHARING_ENABLED_DEFAULT
         );
         if (!isResourceSharingFeatureEnabled) return false;
+        if (request instanceof GetRequest) return false;
```

Comment:
GetRequest needs to be skipped here as that is evaluated as a cluster permission.